tools/SampleAnalyzer/Commons/DataFormat/MCSampleFormat.h

```diff
@@ -63,182 +59,243 @@ namespace MA5
-        std::pair<MAuint32,MAuint32>   beamPDFID_;
-        MAint32                        weightMode_;
-        std::vector<ProcessFormat>     processes_;
-        const MA5GEN::GeneratorType*   sample_generator_;
-        MAfloat32                      length_unit_; /// Length unit: mm=1 cm=0.1
-        MAfloat32                      energy_unit_; /// Energy unit: GeV=1 MeV=0.001 keV=0.000001
+        std::pair<MAint32, MAint32> beamPDGID_;
+        std::pair<MAfloat64, MAfloat64> beamE_;
+        std::pair<MAuint32, MAuint32> beamPDFauthor_;
+        std::pair<MAuint32, MAuint32> beamPDFID_;
+        MAint32 weightMode_;
+        std::vector<ProcessFormat> processes_;
+        const MA5GEN::GeneratorType *sample_generator_;
+        MAfloat32 length_unit_; /// Length unit: mm=1 cm=0.1
+        MAfloat32 energy_unit_; /// Energy unit: GeV=1 MeV=0.001 keV=0.000001
 
         // ----------------------- multiweights ------------------------
+        /// @jackaraz: not sure if this is doing anything
         WeightDefinition weight_definition_;
+        /// @brief store weight names
+        std::map<int, std::string> weight_names_;
 
         // ----------------------- file info ---------------------------
         MAfloat64 xsection_;
         MAfloat64 xsection_error_;
-        MAfloat64 sumweight_positive_;  // all events with positive weights
-        MAfloat64 sumweight_negative_;  // all events with negative weights
-
+        MAfloat64 sumweight_positive_; // all events with positive weights
+        MAfloat64 sumweight_negative_; // all events with negative weights
 
         // -------------------------------------------------------------
         //                      method members
         // -------------------------------------------------------------
-    public :
-
+    public:
         /// Constructor withtout arguments
-        MCSampleFormat(const MA5GEN::GeneratorType* gen)
+        MCSampleFormat(const MA5GEN::GeneratorType *gen)
         {
-            sample_generator_=gen;
+            sample_generator_ = gen;
             Reset();
         }
 
         /// Destructor
-        ~MCSampleFormat()
-        { }
+        ~MCSampleFormat() {}
 
         /// Clear all the content
         void Reset()
         {
             // Physics info
-            beamPDGID_          = std::make_pair(0,0);
-            beamE_              = std::make_pair(0,0);
-            beamPDFauthor_      = std::make_pair(0,0);
-            beamPDFID_          = std::make_pair(0,0);
-            weightMode_         = 0;
+            beamPDGID_ = std::make_pair(0, 0);
+            beamE_ = std::make_pair(0, 0);
+            beamPDFauthor_ = std::make_pair(0, 0);
+            beamPDFID_ = std::make_pair(0, 0);
+            weightMode_ = 0;
             processes_.clear();
             length_unit_ = 1.0;
             energy_unit_ = 1.0;
 
             // WeightDefinition
             weight_definition_.Reset();
```

Comment:
To remove: `weight_definition_.Reset();`.

Reply:
```suggestion
```